lib/web_ui/lib/src/engine/util.dart

```diff
@@ -45,15 +45,27 @@ typedef Callbacker<T> = String? Function(Callback<T> callback);
 ///   return _futurize(_doSomethingAndCallback);
 /// }
 /// ```
+// Note: this function is not directly tested so that it remains private, instead an exact
+// copy of it has been inlined into the test at lib/ui/fixtures/ui_test.dart. if you change
+// this function, then you  must update the test.
 Future<T> futurize<T>(Callbacker<T> callbacker) {
   final Completer<T> completer = Completer<T>.sync();
-  final String? error = callbacker((T t) {
+  // If the callback synchronously throws an error, then synchronously
+  // rethrow that error instead of adding it to the completer. This
+  // prevents the Zone from receiving an uncaught exception.
+  bool sync = true;
```

Comment:
`sync` seems to be a Dart keyword, maybe rename to `isSynchronous`?

https://dart.dev/guides/language/language-tour#keywords

Reply:
Since this is copy-pasta of the mobile version, I'd rather do the clean up later. I added to the list of clean-ups.